docfx_project/docs/com-connection.md

```diff
@@ -0,0 +1,123 @@
+
+`NewSimAutomationConfig.cs`:
+```csharp
+using Cognite.Simulator.Utils.Automation;
+
+public class NewSimAutomationConfig : AutomationConfig
+{
+    public NewSimAutomationConfig()
+    {
+        ProgramId = "Excel.Application";
+    }
+}
+```
+
+Adjust the `ConnectorRuntime` class to use the `NewSimAutomationConfig` class.
+Replace every instance of `AutomationConfig` with `NewSimAutomationConfig`.
+
+```csharp
+using Cognite.Simulator.Utils;
+using Cognite.Simulator.Utils.Automation;
+using CogniteSdk.Alpha;
+using Microsoft.Extensions.DependencyInjection;
+
+public static class ConnectorRuntime {
+
+    public static void Init() {
+        DefaultConnectorRuntime<NewSimAutomationConfig, DefaultModelFilestate, DefaultModelFileStatePoco>.ConfigureServices = ConfigureServices;
+        DefaultConnectorRuntime<NewSimAutomationConfig, DefaultModelFilestate, DefaultModelFileStatePoco>.ConnectorName = "NewSim";
+        DefaultConnectorRuntime<NewSimAutomationConfig, DefaultModelFilestate, DefaultModelFileStatePoco>.SimulatorDefinition = SimulatorDefinition.Get();
+    }
+    static void ConfigureServices(IServiceCollection services)
+    {
+        services.AddScoped<ISimulatorClient<DefaultModelFilestate, SimulatorRoutineRevision>, NewSimClient>();
+    }
+    
+    public static async Task RunStandalone() {
+        Init();
+        await DefaultConnectorRuntime<NewSimAutomationConfig, DefaultModelFilestate, DefaultModelFileStatePoco>.RunStandalone().ConfigureAwait(false);
+    }
+}
+```
+
+### Extend the NewSimClient class to inherit from AutomationClient
+
+This is needed to access the `COM` object.
+We need to add the `using Cognite.Simulator.Utils.Automation;` namespace to the `NewSimClient` class.
+Also, extend the class to inherit from `AutomationClient` and implement a new constructor that calls the base constructor with the AutomationConfig type.
+
+Simply replace these top lines in the `NewSimClient` class:
+
+```csharp
+using Cognite.Simulator.Utils;
+using CogniteSdk.Alpha;
+
+public class NewSimClient : ISimulatorClient<DefaultModelFilestate, SimulatorRoutineRevision>
+{
+// rest of the class
+```
+
+With these lines:
+
+```csharp
+using Cognite.Simulator.Utils;
+using Cognite.Simulator.Utils.Automation;
+using CogniteSdk.Alpha;
+using Microsoft.Extensions.Logging;
+
+public class NewSimClient : AutomationClient, ISimulatorClient<DefaultModelFilestate, SimulatorRoutineRevision>
+{
+    private readonly object simulatorLock = new object();
+    private readonly string _version = "N/A";
+
+    public NewSimClient(ILogger<NewSimClient> logger, DefaultConfig<NewSimAutomationConfig> config)
+            : base(logger, config.Automation)
+    {
+        lock (simulatorLock)
+        {
+            try
+            {
+                Initialize();
+                _version = Server.Version;
+            }
+            finally
+            {
+                Shutdown();
+            }
+        }
+    }
```

Comment:
Lets use semaphoreSlim since its more compatible with Threads statements 

````csharp
private readonly SemaphoreSlim semaphore = new SemaphoreSlim(1, 1);
private string _version = "N/A";

public NewSimClient(ILogger<NewSimClient> logger, DefaultConfig<NewSimAutomationConfig> config)
        : base(logger, config.Automation)
{
    semaphore.Wait();  // Acquire the lock
    try
    {
        Initialize();
        _version = Server.Version;
    }
    finally
    {
        Shutdown();
        semaphore.Release();  // Release the lock
    }
}